integration-commons/src/main/java/org/symphonyoss/integration/utils/RsaKeyUtils.java

```diff
@@ -0,0 +1,162 @@
+   * @return OAuthRsaSigner
+   */
+  public OAuthRsaSigner getOAuthRsaSigner(String privateKey) {
+    OAuthRsaSigner oAuthRsaSigner = new OAuthRsaSigner();
+    oAuthRsaSigner.privateKey = getPrivateKey(privateKey);
+    return oAuthRsaSigner;
+  }
+
+  /**
+   * Convert the passed String into a PrivateKey.
+   * @param privateKey String format of a key.
+   * @return PrivateKey converted.
+   */
+  public PrivateKey getPrivateKey(String privateKey) {
+    byte[] privateBytes = Base64.decodeBase64(privateKey);
+    PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(privateBytes);
+
+    try {
+      KeyFactory kf = KeyFactory.getInstance("RSA");
+      return kf.generatePrivate(keySpec);
+    } catch (NoSuchAlgorithmException e) {
+      throw new UnexpectedAuthException(logMessage.getMessage(RSA_NOT_FOUND), e,
+          logMessage.getMessage(RSA_NOT_FOUND_SOLUTION));
+    } catch (InvalidKeySpecException e) {
+      throw new UnexpectedAuthException(logMessage.getMessage(INVALID_PRIVATE_KEY), e,
+          logMessage.getMessage(INVALID_PRIVATE_KEY_SOLUTION));
+    }
+  }
+
+  /**
+   * Convert the passed String into a PublicKey.
+   * @param publicKey String format of a key.
+   * @return PublicKey converted.
+   */
+  public PublicKey getPublicKey(String publicKey) {
+    try {
+      byte[] encoded = Base64.decodeBase64(publicKey);
+      X509EncodedKeySpec spec = new X509EncodedKeySpec(encoded);
+
+      KeyFactory kf = KeyFactory.getInstance("RSA");
+      return kf.generatePublic(spec);
+    } catch (NoSuchAlgorithmException e) {
+      throw new UnexpectedAuthException(logMessage.getMessage(RSA_NOT_FOUND), e,
+          logMessage.getMessage(RSA_NOT_FOUND_SOLUTION));
+    } catch (InvalidKeySpecException e) {
+      throw new UnexpectedAuthException(logMessage.getMessage(INVALID_PUBLIC_KEY), e,
+          logMessage.getMessage(INVALID_PUBLIC_KEY_SOLUTION));
+    }
+  }
+
+  /**
+   * Remove the prefix and suffix from a String private key.
+   * @param privateKey String private key to be trimmed.
+   * @return Private key String trimmed.
+   */
+  public String trimPrivateKey(String privateKey) {
+    return privateKey.replace(PRIVATE_KEY_PREFIX, StringUtils.EMPTY)
+        .replace(PRIVATE_KEY_SUFFIX, StringUtils.EMPTY);
+  }
+
+  /**
+   * Remove the prefix and suffix from a String public key.
+   * @param publicKey String public key to be trimmed.
+   * @return Public key String trimmed.
+   */
+  public String trimPublicKey(String publicKey) {
+    return publicKey.replace(PUBLIC_KEY_PREFIX, StringUtils.EMPTY)
+        .replace(PUBLIC_KEY_SUFFIX, StringUtils.EMPTY);
+  }
+
+  /**
+   * Gets a public key from a public certificate.
+   * @param certificate X509 certificate to extract the public key from.
+   * @return Extracted Public Key.
+   */
+  public PublicKey getPublicKeyFromCertificate(String certificate) {
+    try {
+      certificate = certificate.replace(X509Factory.BEGIN_CERT, StringUtils.EMPTY);
```

Comment:
Yes, you're right. I usually don't reassign variables in the method, but String in special there is no problem.